### PR TITLE
fix tags in codegen for large field numbers

### PR DESCRIPTION
**Pull request by andrewparmet**
They need to be unsigned in order to be literals.

### CODE REVIEW

File: testing/interop/src/main/proto/google/protobuf/unittest_proto3.proto [hunk 1]
```diff
@@ -0,0 +1,206 @@
...
+}
+
+// TestEmptyMessage is used to test behavior of unknown fields.
+message TestEmptyMessage {}
+
+// TestMessageWithDummy is also used to test behavior of unknown fields.
+message TestMessageWithDummy {
+  // This field is only here for triggering copy-on-write; it's not intended to
+  // be serialized.
+  bool dummy = 536870911;
```
Comment (andrewparmet) [inline testing/interop/src/main/proto/google/protobuf/unittest_proto3.proto:190]: This field breaks generated code. The rest of this file will be used to test the implementation of conversion to DynamicMessage.

File: protokt-runtime/api/protokt-runtime.api [hunk 2]
```diff
@@ -595,7 +595,7 @@ public abstract interface class protokt/v1/KtMessageDeserializer {
 	public abstract fun readSInt32 ()I
 	public abstract fun readSInt64 ()J
 	public abstract fun readString ()Ljava/lang/String;
-	public abstract fun readTag ()I
+	public abstract fun readTag-pVg5ArA ()I
```
Comment (ogolberg) [inline protokt-runtime/api/protokt-runtime.api:598]: what's this?

Comment (andrewparmet) [reply protokt-runtime/api/protokt-runtime.api:598]: UInt nonsense.

Comment (ogolberg) [reply protokt-runtime/api/protokt-runtime.api:598]: is this ABI going to be stable?

Comment (aparmet-toast) [reply protokt-runtime/api/protokt-runtime.api:598]: I don't know. We already do this below - can we merge and revisit pre-1.0?

Comment (andrewparmet) [reply protokt-runtime/api/protokt-runtime.api:598]: I don't know. We already do this below - can we merge and revisit pre-1.0?

**Review by ogolberg — APPROVED** [review]

**Pull request merged**